OpenRA.Mods.Common/Traits/Turreted.cs

```diff
@@ -85,32 +86,30 @@ public static Func<int> TurretFacingFromInit(IActorInitializer init, int def, st
 		}
 
 		public Turreted(ActorInitializer init, TurretedInfo info)
+			: base(info)
 		{
 			this.info = info;
 			TurretFacing = TurretFacingFromInit(init, info.InitialFacing, info.Turret)();
 		}
 
 		public void Created(Actor self)
 		{
-			attack = self.TraitOrDefault<AttackTurreted>();
+			attack = self.TraitsImplementing<AttackTurreted>().ToArray();
```

Comment:
I'm uncertain if I can get that worki9ng in this PR, because if I do it seemingly right, the unit cannot turn any turret which got enabled on the world with an upgrade but was disabled when launch, and I've put 2 days into it without finding the cause of that issue.